hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java

```diff
@@ -1028,9 +1044,24 @@ private void compactIfNecessary(BaseHoodieWriteClient writeClient, String latest
     // let's say we trigger compaction after C5 in MDT and so compaction completes with C4001. but C5 crashed before completing in MDT.
     // and again w/ C6, we will re-attempt compaction at which point latest delta commit is C4 in MDT.
     // and so we try compaction w/ instant C4001. So, we can avoid compaction if we already have compaction w/ same instant time.
-    if (!metadataMetaClient.getActiveTimeline().filterCompletedInstants().containsInstant(compactionInstantTime)
-        && writeClient.scheduleCompactionAtInstant(compactionInstantTime, Option.empty())) {
+    if (metadataMetaClient.getActiveTimeline().filterCompletedInstants().containsInstant(compactionInstantTime)) {
+      LOG.info(String.format("Compaction with same %s time is already present in the timeline.", compactionInstantTime));
```

Comment:
We keep bloating this single class for a long time. We should think about adding a new class called TableServiceMetadateWriter or something and move all table service related code to that. 
I don't want to drag this patch too. but lets see if we can get something in after the major patches are landed.

Reply:
+1
Created https://issues.apache.org/jira/browse/HUDI-6338 to track

Reply:
Already create a ticket you can close either of them as duplicate.